plugin/system_copy.vim

```diff
@@ -22,6 +26,12 @@ function! s:system_copy(type, ...) abort
   echohl String | echon 'Copied to system clipboard via: ' . mode | echohl None
 endfunction
 
+function! s:system_paste() abort
+  let command = <SID>PasteCommandForCurrentOS()
+  put =system(command)
+  echohl String | echon 'Pasted to vim using: ' . command | echohl None
```

Comment:
I suppose you are using Mac and there might be some differences, for me - both on Arch (vim 7.4.778 ) and Ubuntu (vim 7.4.448) this line works as it is expected to.
I like to know what's going on under the hood.
But maybe it's time to decide if this plugin is verbose or silent - or better give this decision to user by making an specific option. Same goes here for custom copy and paste commands. How do you think?